protocols/raft/src/main/java/io/atomix/protocols/raft/storage/snapshot/impl/DefaultSnapshotStore.java

```diff
@@ -205,20 +176,64 @@ public void delete() {
     }
   }
 
-  /** Creates a memory snapshot. */
-  private Snapshot createMemorySnapshot(final DefaultSnapshotDescriptor descriptor) {
-    final HeapBuffer buffer =
-        HeapBuffer.allocate(DefaultSnapshotDescriptor.BYTES, Integer.MAX_VALUE);
-    final Snapshot snapshot = new MemorySnapshot(buffer, descriptor.copyTo(buffer), this);
-    log.debug("Created memory snapshot: {}", snapshot);
-    return snapshot;
+  @Override
+  public PendingSnapshot newPendingSnapshot(
+      final long index, final long term, final WallClockTimestamp timestamp) {
+    return new DefaultPendingSnapshot(newSnapshot(index, term, timestamp));
+  }
+
+  /**
+   * Creates a new snapshot.
+   *
+   * @param index The snapshot index.
+   * @param timestamp The snapshot timestamp.
+   * @return The snapshot.
+   */
+  @Override
+  public Snapshot newSnapshot(
+      final long index, final long term, final WallClockTimestamp timestamp) {
+    final DefaultSnapshotDescriptor descriptor =
+        DefaultSnapshotDescriptor.builder()
+            .withIndex(index)
+            .withTerm(term)
+            .withTimestamp(timestamp.unixTimestamp())
+            .build();
+
+    return createDiskSnapshot(descriptor);
+  }
+
+  @Override
+  public void purgeSnapshots(final Snapshot snapshot) {
```

Comment:
Who calls this?

Reply:
Is this then necessary to have this in the interface?

Reply:
This is called by Zeebe and is sort of necessary for our implementation. We can discuss it over slack or face to face